walk/config.go

```diff
@@ -104,6 +112,46 @@ func (cr *Configurer) Configure(c *config.Config, rel string, f *rule.File) {
 	c.Exts[walkName] = wcCopy
 }
 
+func (c *Configurer) loadBazelIgnore(repoRoot, rel string, wc *walkConfig) error {
+	// Only load .bazelignore at the root, and only once
+	if rel != "" {
+		return nil
+	}
+
+	ignorePath := path.Join(repoRoot, ".bazelignore")
+	file, err := os.Open(ignorePath)
+	if os.IsNotExist(err) {
+		return nil
+	}
+	if err != nil {
+		return fmt.Errorf(".bazelignore exists but couldn't be read: %v", err)
+	}
+	defer file.Close()
+
+	wc.excludes = append(wc.excludes, path.Join(rel, ".bazelignore"))
+
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		ignore := strings.TrimSpace(scanner.Text())
+		if ignore == "" || string(ignore[0]) == "#" {
+			continue
+		}
+		// Bazel ignore paths are always relative to repo root.
+		// Glob patterns are not supported.
+		pat := path.Join(rel, ignore)
+		if isGlobPattern(pat) {
+			log.Printf("the .bazelignore exclusion pattern must not be a glob %s", pat)
+			continue
+		}
```

Comment:
nit: we only need to check for glob pattern in `ignore` and skip it early?

Reply:
agreed but going to keep the log as it's a better DX